File: Backend/src/main/java/com/Exam/Backend/Service/questionService.java
package com.Exam.Backend.Service;

import com.Exam.Backend.Model.Exam.Question;
import com.Exam.Backend.Model.Exam.Quiz;

import java.util.List;

public interface questionService {

    public Question addQuestion(Question question);
    public Question updateQuestion(Long id,Question question) throws Exception;
    public List<Question> getAllQuestions();
    public Question getQuestionById(Long qId);
    public void deleteQuestion(Long qId);
    public List<Question> getQuestionsOfQuiz(Quiz quiz);
    public Question get(Long qid);



}

File: Frontend/src/app/services/question.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class QuestionService {

  constructor(private http:HttpClient) { }


  public getQuestionsOfQuiz(qid:any){
    const token = localStorage.getItem('token');
    const headers = new HttpHeaders().set('Authorization', `Bearer ${token}`);
    return this.http.get(`${baseUrl}/question/quiz/all/${qid}`,{headers});
  }

  public getQuestionsOfQuizForUser(qid:any){
    const token = localStorage.getItem('token');
    const headers = new HttpHeaders().set('Authorization', `Bearer ${token}`);
    return this.http.get(`${baseUrl}/question/quiz/${qid}`,{headers});
  }

  public addQuestion(question:any){
    const token = localStorage.getItem('token');
    const headers = new HttpHeaders().set('Authorization', `Bearer ${token}`);
    return this.http.post(`${baseUrl}/question/`,question,{headers});
  }

  public deleteQuestion(qid:any){
    const token = localStorage.getItem('token');
    const headers = new HttpHeaders().set('Authorization', `Bearer ${token}`);
    return this.http.delete(`${baseUrl}/question/${qid}`,{headers});
  }

  public evalQuiz(questions:any){
    const token = localStorage.getItem('token');
    const headers = new HttpHeaders().set('Authorization', `Bearer ${token}`);
    return this.http.post(`${baseUrl}/question/eval-quiz`,questions,{headers});
  }
}
